packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart

```diff
@@ -404,6 +453,130 @@ class ResidentWebRunner extends ResidentRunner {
     return OperationResult.ok;
   }
 
+  Future<int> _runExperimental({
+    Completer<DebugConnectionInfo> connectionInfoCompleter,
+    Completer<void> appStartedCompleter,
+    String route,
+    ApplicationPackage applicationPackage,
+  }) async {
+    final String effectiveHostname = debuggingOptions.hostname ?? 'localhost';
+    final int hostPort = debuggingOptions.port == null
+      ? await os.findFreePort()
+      : int.tryParse(debuggingOptions.port);
+    device.devFS = WebDevFS(
+      effectiveHostname,
+      hostPort,
+      packagesFilePath,
+    );
+    await device.devFS.create();
+    await _updateDevFS(fullRestart: true);
+    device.generator.accept();
+    await device.device.startApp(applicationPackage,
+      mainPath: target,
+      debuggingOptions: debuggingOptions,
+      platformArgs: <String, Object>{
+        'uri': 'http://$effectiveHostname:$hostPort',
+      },
+    );
+    return attach(
+      connectionInfoCompleter: connectionInfoCompleter,
+      appStartedCompleter: appStartedCompleter,
+    );
+  }
+
+  /// Different implementations for the incremental compiler based workflow.
+  Future<int> _attachExperimental({
+    Completer<DebugConnectionInfo> connectionInfoCompleter,
+    Completer<void> appStartedCompleter,
+  }) async {
+    final Chrome chrome = await ChromeLauncher.connectedInstance;
+    final ChromeTab chromeTab = await chrome.chromeConnection.getTab((ChromeTab chromeTab) {
+      return chromeTab.url.contains(debuggingOptions.hostname);
+    });
+    _wipConnection = await chromeTab.connect();
+    appStartedCompleter?.complete();
+    connectionInfoCompleter?.complete();
+    final int result = await waitForAppToFinish();
+    await cleanupAtFinish();
+    return result;
+  }
+
+  Future<OperationResult> _restartExperimental({
+    bool fullRestart = false,
+    bool pauseAfterRestart = false,
+    String reason,
+    bool benchmarkMode = false,
+  }) async {
+    final UpdateFSReport report = await _updateDevFS(fullRestart: fullRestart);
+    if (report.success) {
+      device.generator.accept();
+    } else {
+      await device.generator.reject();
+    }
+    final String modules = report.invalidatedModules
+      .map((String module) => '"$module"').join(',');
+    if (fullRestart) {
+      await _wipConnection.sendCommand('Page.reload');
+    } else {
+      try {
+        await _wipConnection.debugger
+          .sendCommand('Runtime.evaluate', params: <String, Object>{
+          'expression': 'window.\$hotReloadHook([$modules])',
+          'awaitPromise': true,
+          'returnByValue': true,
+        });
+      } on WipError catch (err) {
+        printError(err.toString());
+        return OperationResult(1, '');
```

Comment:
How about put `err.toString()` as the message in `OperationResult`?

Reply:
Done